docs/DocGenerator/Config.py

```diff
@@ -0,0 +1,127 @@
+from collections import namedtuple
+import yaml
+
+CONFIG_PATH = "config.yaml"
+
+
+class Config():
+    def __init__(self):
+        self.project_path = "../.."
+        self.documentation_path = ".."
+        self.include_paths = []
+        self.include_regex = []
+        self.ignore_paths = []
+        self.valid_tags = []
+        self.required_fields = self.__fields()
+        self.ignored_fields = self.__fields()
+        with open(CONFIG_PATH) as fd:
+            self.__config_data = yaml.load(fd)
+
+        self.__read_project_path()
+        self.__read_documentation_path()
+        self.__read_include_paths()
+        self.__read_include_regex()
+        self.__read_ignore_paths()
+        self.__read_valid_tags()
+        self.__read_required_fields()
+        self.__read_ignored_fields()
+
+    def __read_project_path(self):
+        if 'Project path' in self.__config_data:
+            self.project_path = self.__config_data['Project path']
+
+    def __read_documentation_path(self):
+        if 'Documentation path' in self.__config_data:
+            self.documentation_path = self.__config_data['Documentation path']
+
+    def __read_include_paths(self):
+        if not 'Include paths' in self.__config_data:
+            raise Exception("Include paths are empty")
+        for include in self.__config_data['Include paths']:
+            if not 'path' in include:
+                raise Exception("One include path is missing")
+            element = self.__paths()
+            element.path = include['path']
+            if 'recursive' in include:
```

Comment:
How would be the behavior if there's not a `recursive` option?

Reply:
Without `recursive` option, the parser will only look for the files at the the first level of this path.